solutions/fully-configurable/variables.tf

```diff
@@ -0,0 +1,492 @@
+##############################################################################
+# Input Variables
+##############################################################################
+
+variable "ibmcloud_api_key" {
+  type        = string
+  description = "The IBM Cloud API key to deploy resources."
+  sensitive   = true
+}
+
+variable "provider_visibility" {
+  description = "Set the visibility value for the IBM terraform provider. Supported values are `public`, `private`, `public-and-private`. [Learn more](https://registry.terraform.io/providers/IBM-Cloud/ibm/latest/docs/guides/custom-service-endpoints)."
+  type        = string
+  default     = "public"
+
+  validation {
+    condition     = contains(["public", "private", "public-and-private"], var.provider_visibility)
+    error_message = "Invalid visibility option. Allowed values are 'public', 'private', or 'public-and-private'."
+  }
+}
+
+variable "existing_resource_group_name" {
+  type        = string
+  description = "The name of an existing resource group to provision the resources."
+}
+
+variable "prefix" {
+  type        = string
+  nullable    = true
+  description = "Prefix to add to all the resources created by this solution. To not use any prefix value, you can set this value to `null` or an empty string."
+}
+
+variable "vpc_name" {
+  default     = "vpc"
+  description = "Name of the VPC. If a prefix input variable is specified, the prefix is added to the name in the `<prefix>-<name>` format."
+  type        = string
+}
+
+variable "region" {
+  default     = "us-south"
+  description = "Region to deploy the VPC."
+  type        = string
+}
+
+variable "resource_tags" {
```

Comment:
different vpc resource support tags (e.g. a VPC can have different tags to subnets). I think this needs to be broken down. VPC tags variable should be explicitly named `vpc_resource_tags`

Reply:
As discussed, subnet_tags has been added explicitly.